corehq/apps/app_manager/suite_xml/sections/entries.py

```diff
@@ -981,12 +981,15 @@ def append_update(dict_, new_dict):
                     if not parent_datum_meta.requires_selection:
                         # Add parent datums of opened subcases and automatically-selected cases
                         datums.insert(index, attr.evolve(parent_datum_meta, from_parent=True))
-                    elif this_datum_meta and this_datum_meta.case_type == parent_datum_meta.case_type:
-                        append_update(changed_ids_by_case_tag,
-                                      rename_other_id(this_datum_meta, parent_datum_meta, datum_ids))
-                        append_update(changed_ids_by_case_tag,
-                                      get_changed_id(this_datum_meta, parent_datum_meta))
-                        this_datum_meta.datum.id = parent_datum_meta.datum.id
+                    elif this_datum_meta:
+                        same_case_type = this_datum_meta.case_type == parent_datum_meta.case_type
+                        same_datum_type = type(this_datum_meta.datum) == type(parent_datum_meta.datum)
```

Comment:
I sorta agree with stickler on this one, in that I'm not certain what the business logic check here is.  What should be unique? Should it compare `datum_meta.ROOT_NAME`?

Reply:
updated to use `ROOT_NAME`